src/lib/mina/v2/transaction.ts

```diff
@@ -1,78 +1,133 @@
 import { Memo } from '../../../mina-signer/src/memo.js';
 import { hashWithPrefix, prefixes } from '../../../mina-signer/src/poseidon-bigint.js';
 import { Signature, signFieldElement } from '../../../mina-signer/src/signature.js';
 import { NetworkId } from '../../../mina-signer/src/types.js';
+import { LedgerView, ChainView } from './views.js';
+import { PublicKey } from 'src/lib/provable/crypto/signature.js';
 
-export { ZkappCommand, ZkappFeePayment, ZkappCommandContext, AuthorizedZkappCommand };
+export { ZkappCommand, FeePayerAccountUpdate, ZkappCommandContext };
 
-interface ZkappFeePaymentDescription {
-  publicKey: PublicKey;
-  fee: UInt64; // TODO: abstract currency representation
-  validUntil?: UInt32; // TODO: abstract slot representation
-  nonce: UInt32;
+interface FeePayerInputs {
+  readonly publicKey: PublicKey;
+  readonly fee: UInt64; // TODO: abstract currency representation
+  readonly validUntil?: UInt32; // TODO: abstract slot representation
+  readonly nonce: UInt32;
 }
 
-class ZkappFeePayment {
-  readonly __type: 'ZkappCommand' = 'ZkappCommand' as const;
+interface ZkappCommandInputs {
+  feePayer: FeePayerInputs | FeePayerAccountUpdate;
+  // TODO: merge AccountUpdateTree and AccountUpdate
+  accountUpdates: (AccountUpdate | AccountUpdateTree<AccountUpdate>)[];
+  memo?: string;
+}
+
+interface Commitments {
+  accountUpdateForestCommitment: BigInt;
+  fullTransactionCommitment: BigInt;
+}
+
+// TODO: Are there actually any other types of transaction?  Or only ZkAppCommand?
+interface Transaction<T extends Transaction<T>> {
+  readonly __type: ['SignedCommand', 'ZkAppCommand'][number];
+
+  /**
+   * Short string (max length 32) that can be used to describe the transaction.
+   * This is not used in the protocol, but it is public on chain and can be used
+   * for indexing or in application logic.
+   *
+   * TODO: we probably want an explicit memo type instead to help enforce these rules early and not surprise the user when their memo changes slightly later
+   */
+  memo?: string;
+
+  /**
+   * The public key of the account, the amount, and the metadata associated with the fee for this transaction.
+   */
+  feePayerAccountUpdate: FeePayerAccountUpdate;
+
+  isAuthorized: boolean;
+
+  /**
+   * Commits to the state of the transaction so that Mina can verify the authorization
+   *
+   * @param networkId The id of the Mina network (e.g. 'testnet', 'mainnet')
+   * @returns The Field commitments to the account update forest and the full transaction as bigints
+   */
+  commitments(networkId: NetworkId): Commitments;
+
+  /**
+   * Authorizes all account updates in the transaction, either by signature or by proof
+   *
+   * @returns An authorized {@link Transaction}
+   */
+  authorize(...args: any): T | Promise<T>;
+
+  /**
+   * Converts the transaction to a JSON object.
+   *
+   * @returns A JSON object representing the transaction
+   */
+  toJSON(): any;
```

Comment:
`any` could be replaced here by a `SerializedMinaTransaction` type that helps map to the raw transaction JSON that Mina nodes expect.